commit/merkleroot/processor.go

```diff
@@ -0,0 +1,83 @@
+package merkleroot
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+	"github.com/smartcontractkit/chainlink-ccip/shared"
+)
+
+// Processor is the processor responsible for
+// reading next messages and building merkle roots for them,
+// It's setup to use RMN to query which messages to include in the merkle root and ensures
+// the newly built merkle roots are the same as RMN roots.
+type Processor struct {
+	nodeID       commontypes.OracleID
+	cfg          pluginconfig.CommitPluginConfig
+	lggr         logger.Logger
+	observer     Observer
+	readerSyncer *plugincommon.BackgroundReaderSyncer
+	ccipReader   reader.CCIP
+	reportingCfg ocr3types.ReportingPluginConfig
+	chainSupport shared.ChainSupport
+}
+
+// NewProcessor creates a new Processor
+func NewProcessor(
+	nodeID commontypes.OracleID,
```

Comment:
Is this unused, maybe it can be removed?
```suggestion
	oracleID commontypes.OracleID,
```